### PR TITLE
SLING-12052: add Pagefind for site search

### PR DESCRIPTION
**Pull request by bdelacretaz**
This PR adds a search box at the top of the pages, using Pagefind.

Could someone review it before I merge it? 

To test it on your local box use

    mvn clean package -Ppagefind,run-site

And open http://localhost:8820/

### CODE REVIEW

**Comment by kwin** [comment]
We probably need to slightly tweak the markup to consider some head elements (like title) for the search index: https://pagefind.app/docs/metadata/. Otherwise only the body content counts IIUC.

File: src/main/jbake/templates/layout/main.tpl [hunk 1]
```diff
@@ -24,7 +24,7 @@ html(lang:'en'){
 
                     if( content ) {
                         if(content.title) {
-                            h1(class:"title") {
+                            h1(class:"title","data-pagefind-body":true) {
```
Comment (kwin) [inline src/main/jbake/templates/layout/main.tpl:27]: I am not sure this changes anything, as this is already considered (as below body, https://pagefind.app/docs/indexing/#limiting-what-sections-of-a-page-are-indexed). What about increasing the ranking with data-pagefind-weight="10" (or "9") for this particular h1?

**Comment by bdelacretaz** [comment]
It looks like pagefind does not index a title that it considers metadata, that's just used when displaying the list of results.

I have now added a `display:none;` div with the title at a weight of 7.0 and it looks like titles are indexed now.

However, searching for "sling models" still doesn't return the models page as the first result, I don't know why.

File: src/main/jbake/templates/layout/main.tpl (hunk 1 above)
Comment (bdelacretaz) [reply src/main/jbake/templates/layout/main.tpl]: > Wouldn't this restrict indexing to this title only per page...

The docs that you point to say "Multiple data-pagefind-body elements may exist on a page, and their content will be combined" - but this didn't work, it looks like pagefind does not index what it finds to be a title, due to it being the first H1 on the page. I have removed this attribute now.

**Comment by bdelacretaz** [comment]
I think this is good enough to commit as a first version, planning to merge this PR early next week unless someone objects.

**Pull request merged**